pkg/datastore/test/revisions.go

```diff
@@ -125,13 +127,16 @@ func RevisionGCTest(t *testing.T, tester DatastoreTester) {
 	require.NoError(err)
 	require.NoError(ds.CheckRevision(ctx, head), "expected head revision to be valid in GC Window")
 
-	// Make sure GC kicks in after the window.
-	time.Sleep(300 * time.Millisecond)
```

Comment:
would this work with 600 ms instead? it seems like the flakiness could be related to jitter?

Reply:
The flakiness was around the GC process itself running, rather than the window having passed. So long as we have *at least* 300ms, we should be fine (since it is the window size)

Reply:
I'm still concerned about this - if the GC window is 300ms and we're waiting for _10s_ and it's not running again, something seems off somewhere?

Reply:
One thing that might help illuminate - what if we change the `hasGCRun` bool to an atomic uint that we increment on each GC run?

The test can print that value in the eventually check and we can see if GC just isn't running or if there's some issue with marking it as having run.

Reply:
It is incredibly hard to reproduce locally, but I suspect the problem is that the GC is encountering errors on the locking because the PG container is overloaded. This, in turn, causes the GC runner to continuously "push out" running the GC process (because of the exponential backoff). However, waiting for the *background* process to run is not the point of the test, so I changed it to always run it directly.

We could add a distinct test to ensure the GC process itself runs once, without having it run at an interval of `10ms` (which is what the test had here previously), but we have distinct GC tests for those in each datastore, so I think it is fine to change the generic one to just run it manually

Reply:
What other test asserts that the background GC process actually runs? having trouble finding it.

Reply:
MySQL appears to have it but Postgres does not. I can add a test to the PG driver if you like or add one that just does a wait-and-see to see if GC runs at all in the generic tests